src/AI/ModularAI.java

```diff
@@ -161,13 +161,13 @@ public GameAction getUnitAction(Unit unit, GameMap map)
     }
   }
 
-  public static abstract class UnitActionFinder implements AIModule
+  public static abstract class UnitActionFinder<T> implements AIModule
```

Comment:
> Throw the CRTP on UnitActionFinder

That doesn't look like CRTP - `UnitActionFinder` inherits from `AIModule`, but is templated with `ModularAI`. In CRTP, child classes template their base classes with themselves. Or am I misreading something?
I'm also not sure what we gain from this template in the first place, since it's just storing the `ModularAI` in a variable.

Reply:
Okay, fair. It's not exactly CRTP - it's the parent class's owned/static class that is templated with the child class in this case. It's CRTP-adjacent if you consider the parent's owned class(es) to be entangled with the parent class in a relevant way.

I gain... like 3 less lines of code per `ActionFinder` definition. Which probably isn't worth it, but I hate boilerplate.